Weight loss program

samanpwbb opened the pull request:
This is ready to go. Decided against drastic conceptual changes.

// mb-pages
CSS: 262 kB (minified) => 29.7 kB (gzipped)
SVG: 50 kB => 15.1 kB (gzipped)

// This branch
CSS: 172 kB (minified) => 21.9 kB (gzipped)
SVG: 50 kB => 15.1 kB (gzipped)

---

This branch turned into a larger v1 push, beyond just space saving. Changes:

- Adds missing --dark modifiers to some typography rules.
- Stroked buttons and select elements now have a 1px stroke by default.
- Adds new `btn--stroked--2` and `select--stroked--2` modifier classes to create stroked buttons and select elements that have 2px strokes.
- [internal only] Renames `neutral` variables to more meaningful `disabled-{type}-interactive` variables.
- Darken the `gray-faint` color to #e5e5e5 so it's legible on a wider range of monitors.
- Removes teal color variations.
- Removes 600px layout scale value for padding and margin.
- Refactors range inputs to use currentColor for coloring, which save space
- Removes `darken5` and `lighten5` color variations to save space and encourage more accessible design. 
- Removes `p{n}` and `m{n}` rules.
- Limits the possible colors available for form elements for accessibility reasons and to save size. `-faint` or `-light` form elements are no longer available for _any element_, and `darken10` and `darken10` are no longer available for any for element except buttons.
- Adds distinct hover states for active checkboxes and radio inputs.
- Redesigns checkbox according to #745.

samanpwbb committed:
simplify color variants

samanpwbb committed:
reduce filesize by trimming color and layout rules, closes #746

samanpwbb committed:
fix rules based on scale changes

samanpwbb committed:
remove lighten5 and darken5, to save space and encourage accessible styling

samanpwbb committed:
minor cleanup

samanpwbb committed:
site adjustments, make faint gray legible

samanpwbb committed:
fix stroked buttons

samanpwbb committed:
cleanup select elements

samanpwbb committed:
add 1px buttons and pills, closes #747

samanpwbb committed:
refine checkbox and radio styles, closes #745

samanpwbb committed:
remove unecessary check in isNotSuitableForForms

samanpwbb committed:
fix select element across all browsers

samanpwbb committed:
update jest snapshots

samanpwbb commented:
@tristen @davidtheclark this is ready for review.

tristen commented on site/catalog/toggles.js:
```diff
@@ -84,12 +78,12 @@ class Toggles extends React.Component {
 
         <div className='mb12 txt-bold color-darken50 txt-uppercase txt-s'>Light variations</div>
-        <div className='bg-gray round p12y mt12'>
+        <div className='bg-gray round px12 py12y mt12'>
```
Should be `py12`

tristen commented on site/catalog/toggles.js:
```diff
@@ -84,12 +78,12 @@ class Toggles extends React.Component {
 
         <div className='mt24 mb12 txt-bold color-darken50 txt-uppercase txt-s'>Disabled and light</div>
-        <div className='bg-gray round p12y mt12'>
+        <div className='bg-gray round px12 py12y mt12'>
```
Should be `py12`

tristen commented on site/examples/progress_bars.js:
```diff
@@ -9,7 +9,7 @@ const playback = `<div class='flex-parent flex-parent--row flex-parent--center-c
       <div class='triangle--r triangle color-green color-green'></div>
     </button>
-    <div class='flex-child flex-child--grow p18 bg-white'>
+    <div class='flex-child flex-child--grow px18 py18bg-white'>
```
Missing space between class names

tristen commented on site/examples/tooltips.js:
```diff
@@ -2,22 +2,22 @@ import React from 'react';
 const basic_right = `<div class='flex-parent-inline flex-parent--center-cross'>
   <span class='flex-child triangle triangle--l'></span>
-  <div class='flex-child p6 round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
+  <div class='flex-child px6 py6round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
```
Missing space between class names

tristen commented on site/examples/tooltips.js:
```diff
@@ -2,22 +2,22 @@ import React from 'react';
 
 const basic_left = `<div class='flex-parent-inline flex-parent--center-cross'>
-  <div class='flex-child p6 round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
+  <div class='flex-child px6 py6round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
```
Missing space between class names

tristen commented on site/examples/tooltips.js:
```diff
@@ -2,22 +2,22 @@ import React from 'react';
 
 const basic_with_closure = `<div class='flex-parent-inline flex-parent--center-cross flex-parent--column'>
-  <div class='flex-child relative p18 round txt-bold txt-s align-center bg-darken75 color-white'>
+  <div class='flex-child relative px18 py18round txt-bold txt-s align-center bg-darken75 color-white'>
```
Missing space between class names

tristen commented on src/typography.css:
```diff
@@ -39,13 +38,10 @@ textarea {
-  border: 1px solid var(--lighten10);
-  line-height: 18px;
-  border-radius: 3px;
-  padding: 2px 3px;
```
Was dropping these properties intentional?

samanpwbb replied:
Yeah, prose--dark is a modifier for prose so no need to duplicate these rules.

tristen commented on src/variables.json:
```diff
@@ -60,9 +53,9 @@
-  "neutral45": "rgba(127, 127, 127, 0.45)",
+  "disabled-primary-interactive-color": "rgba(127, 127, 127, 0.25)",
+  "disabled-secondary-interactive-color": "rgba(127, 127, 127, 0.1)",
+  "disabled-text-color": "rgba(127, 127, 127, 0.45)",
```
Is `disabled` accurate? This is used for placeholder too right? And I'm not sure how `primary` & `secondary` align with how they are used. How about `input-{value}-on-blur`?

samanpwbb replied:
Ohh that is the one inaccurate use. I can change this to be 'placeholder-text-color'

tristen commented on site/catalog/toggles.js:
```diff
@@ -2,27 +2,21 @@ import React from 'react';
-  'gray-faint',
-  'gray-light',
   'gray',
-  'gray-dark',
```
Curious about all these `gray-{varient}` omissions? They ... still exist but you're removing them from the catalog?

samanpwbb replied:
This was a fix - we never had -dark variants here.

tristen reviewed (changes requested):
Changes look great!

> Decided against drastic conceptual changes

By that do you mean reducing color variations on form elements to two states?

samanpwbb committed:
address PR feedback
- rename disabled-text-color to more appropriate inactice-text-color, used for both disabled and placeholder text
- fix find and replace typos

samanpwbb commented:
Addressed feedback!

davidtheclark commented on scripts/build-color-variants.js:
```diff
@@ -52,13 +47,11 @@ const allColors = [
   'darken50',
   'darken75',
 
-  'lighten5',
```
I like these cuts!

davidtheclark commented on scripts/build-color-variants.js:
```diff
@@ -73,8 +66,15 @@ function isSemitransparent(color) {
+function isNotSuitableForButtons(color) {
+  // Filled Buttons should be allowed to have subtle backgrounds within reason.
+  return color === 'black' || /(-faint|-dark)$/.test(color);
 }
```
🎉

davidtheclark commented on scripts/build-color-variants.js:
```diff
@@ -300,51 +281,59 @@ function buildColorVariants(variables, config) {
     }, '');
   };
 
   variantGenerators.range = function (colors) {
```
Dark and non-dark ranges seem to look the same:

![screen shot 2017-06-04 at 8 22 01 am](https://cloud.githubusercontent.com/assets/628431/26762900/e8ded1b8-48fe-11e7-9b30-a7a8e67fb879.png)

samanpwbb replied:
Good catch - was from an earlier refactor attempt. I removed this class but didn't remove the documentation?

samanpwbb replied:
Done!

davidtheclark commented on scripts/build-color-variants.js:
```diff
@@ -73,8 +66,15 @@ function isSemitransparent(color) {
+  // Do not create form elements from values that are too light or too dark,
+  // for accessibility and to save space.
+  return color === 'black' || /^(darken10|lighten10)$/.test(color) || /(-dark|-light|-faint)$/.test(color);
+}
```
What do you about think `-faint` for hover and active states? Should we keep those, or also remove them to encourage stronger contrast?

samanpwbb replied:
I'd be in favor of cutting them - can add that to this PR

davidtheclark commented:
Since a lot changes here, might be useful to add a Changelog modifications in this PR.

samanpwbb committed:
remove outdated documentation

samanpwbb committed:
reduce options for on-hover and on-active to save space/encourage accessibility

samanpwbb committed:
update changelog with to-be-released changes

samanpwbb commented:
@tristen @davidtheclark oookay i think this is ready for a final 👀  then merge

davidtheclark reviewed (approved):
All these changes seem wonderful to me! Very fine work.
